src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -2957,7 +2957,7 @@ public UsingStatementAst(IScriptExtent extent, HashtableAst moduleSpecification)
         /// <param name="aliasName">The name of the alias.</param>
         /// <param name="resolvedAliasAst">The item being aliased.</param>
         public UsingStatementAst(IScriptExtent extent, UsingStatementKind kind, StringConstantExpressionAst aliasName,
-                                 StringConstantExpressionAst resolvedAliasAst)
+                                 Ast resolvedAliasAst)
```

Comment:
This as well as the property type change would be a binary breaking change. I know it doesn't currently work, but this would cause a JIT time error for anyone trying to plan ahead (even as a more generic type, the signature change will cause the method to no longer be found).

I'll need to think about how to best solve this. I'm leaning towards:

1. Keep the existing method and property
2. Mark them `Obsolete`
3. Throw a semantic check error the obsolete property is populated (e.g. via a manually created AST)
4. Create a new ctor and property that takes the specific desired AST type (presumably `TypeExpressionAst`) and name it `TypeAlias` (or similar)

Reply:
I've changed it back to a StringConstantExpression and I'm now scanning the text in the few places where I need the typename. See this commit: https://github.com/PowerShell/PowerShell/pull/16734/commits/300efecb924fcef9f1409e7837eee23a0a23c25e  
Is that fine?

Reply:
So thinking from the perspective of an AST consumer I think it'd be better to have a separate property (so if I want to resolve the type in some tool I don't have to parse it myself). But good call on reverting this back to `StringConstantExpressionAst`, I think adding a separate property for `TypeAlias` that is `TypeExpressionAst` should be 💯

Reply:
Honestly I'm unsure that `using command X = Y` makes sense as a feature, so I'm leaning towards saying we should cross that bridge if we come to it.

For `using namespace x = y` my understanding was that that's what `StringConstantExpression` would represent, and `using type x = y` would be the `TypeAlias`. Is that incorrect?

Reply:
> For using namespace x = y my understanding was that that's what StringConstantExpression would represent, and using type x = y would be the TypeAlias. Is that incorrect?

No, I just wanted to double check that this was the idea. (Your original plan indicated that you wanted to mark the `Alias` property as Obsolete, but that wouldn't make sense if we plan on continuing to use it for namespace aliases.).